k9mail-library/src/main/java/com/fsck/k9/mail/Message.java

```diff
@@ -117,8 +117,32 @@ public void setRecipient(RecipientType type, Address address) throws MessagingEx
 
     public abstract void setReferences(String references) throws MessagingException;
 
+    @Override
+    public abstract Body getBody();
+
+    @Override
+    public abstract String getContentType() throws MessagingException;
+
+    @Override
+    public abstract void addHeader(String name, String value) throws MessagingException;
+
+    @Override
+    public abstract void addRawHeader(String name, String raw) throws MessagingException;
+
+    @Override
+    public abstract void setHeader(String name, String value) throws MessagingException;
+
+    @Override
+    public abstract String[] getHeader(String name) throws MessagingException;
+
     public abstract Set<String> getHeaderNames() throws MessagingException;
 
+    @Override
+    public abstract void removeHeader(String name) throws MessagingException;
+
+    @Override
+    public abstract void setBody(Body body) throws MessagingException;
```

Comment:
I don't remember. I guess it was done for consistency reasons.
But I think removing all the interface methods from `Message` is a good idea. I see no added value.